### ISSUE TITLE
Incorrect XML Layout appears with Appium Inspector for react native android App on Samsung device

### ISSUE THREAD

**Issue by bipin-k** [appium/appium-inspector#731]
## Appium or Appium Desktop?

- [x] I have verified that my issue does not occur with Appium and should be investigated as an Appium Desktop issue

## The problem

When click on `Refresh Source & Screenshot` button at appium inspector, the xml layout for the `Set Language to English` is coming differently on Samsung device where as the same XML Layout on OnePlus & Pixel devices are coming as expected. The issue for xml layout on Samsung device is inconsistent on what is fetched via appium and on real device as shown on screenshot. Why there is such discrepancy with capatured XML Layout with Samsung device and not with OnePlus and Pixel devices.
NOTE: The app as shown in reference is build with react native.

## Environment

- I am running Appium Desktop version v2023.2.1.
- I am on (pick one):
    - [x] Mac

![Image 28-02-23 at 5 02 PM (3)](https://user-images.githubusercontent.com/21119358/221842298-d7a612fe-5ebb-4220-8494-ffaa8bc60ff0.jpeg)
![Image 28-02-23 at 5 02 PM (2)](https://user-images.githubusercontent.com/21119358/221842327-fde27f7c-3b1b-4f00-8f6e-c5706f1e1a48.jpeg)
![Image 28-02-23 at 5 02 PM](https://user-images.githubusercontent.com/21119358/221842336-4f8e4c10-dea5-4043-ac84-3bf40bcda80a.jpeg)

**Comment by bipin-k** [comment]
Also, the layout of the UI as when selected at the root tag of the XML Layout, the resolution of the selected UI on device is way smaller than the UI as shown on the real device app as per
![Image 28-02-23 at 5 37 PM](https://user-images.githubusercontent.com/21119358/221850703-9fd9046c-d778-4422-94f1-b239f776cf62.jpeg)
 shown on the below screenshot attached.

**Comment by jlipps** [comment]
what is the display size as reported to appium by the different devices? maybe the samsung device is not reporting its size or pixel density appropriately?

**Comment by bipin-k** [comment]
@jlipps yes, the output with `driver.manage().window().getSize()` differs from `adb shell wm size` as when executed with below script with appium,

```
package live.mpl.appium;

import io.appium.java_client.AppiumDriver;
import io.appium.java_client.android.AndroidDriver;
import java.net.MalformedURLException;
import java.net.URL;
import java.time.Duration;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class MobileScreenSizeTest {

  private AppiumDriver driver;

  @BeforeMethod
  public void setup() throws MalformedURLException {
    DesiredCapabilities androidCaps = new DesiredCapabilities();
    androidCaps.setCapability("appPackage", "com.mpl.androidapp");
    androidCaps.setCapability("appActivity", "com.mpl.androidapp.MPLSplashActivity");

    androidCaps.setCapability("automationName", "UIAutomator2");
    androidCaps.setCapability("platformName", "Android");

    driver = new AndroidDriver(new URL("http://127.0.0.1:4724/wd/hub"), androidCaps);
    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(60));
  }

  @Test
  public void displaySize() {
    System.out.println("Dimension --> " + driver.manage().window().getSize());
    Map<String, Object> args = new HashMap<>();
    args.put("command", "wm");
    args.put("args", List.of("size"));
    String output = driver.executeScript("mobile: shell", args).toString();
    System.out.println("ADB --> " + output);
  }

  @AfterMethod
  public void teardown() {
    driver.quit();
  }
}
```

`OUTPUT`
```
Dimension --> (1080, 2131)
ADB --> Physical size: 1080x2340
```

**Comment by bipin-k** [comment]
@KazuCocoa @SrinivasanTarget @saikrishna321 ^^

**Comment by KazuCocoa** [comment]
Potentially newer `androidx.test.uiautomator:uiautomator` will help as https://github.com/appium/appium-uiautomator2-server/issues/490
The latest uia package version is still alpha, so UIA2 server won't have it yet, but you can try the version out on your local.

**Comment by eglitise** [comment]
Closing due to no response
